staging/src/k8s.io/cli-runtime/pkg/resource/dry_run_verifier.go

```diff
@@ -73,12 +74,37 @@ func hasGVKExtension(extensions []*openapi_v2.NamedAny, gvk schema.GroupVersionK
 type DryRunVerifier struct {
 	finder        CRDFinder
 	openAPIGetter discovery.OpenAPISchemaInterface
+	oapiLock      sync.RWMutex
+	oapi          *openapi_v2.Document
+}
+
+func (v *DryRunVerifier) openAPISchema() (*openapi_v2.Document, error) {
+	if schema := func() *openapi_v2.Document {
```

Comment:
I think it's better to use sync.Once like https://github.com/kubernetes/kubernetes/blob/84483a5aacba8e535c9e185e9b0c01a5248b5086/staging/src/k8s.io/kubectl/pkg/cmd/util/factory_client_access.go#L45

Reply:
That one isn't particularly great in case of errors, as it will at most be called once